src/Datadog.Trace/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Newtonsoft.Json.Linq;
+
+namespace Datadog.Trace.Plugins
+{
+    internal static class PluginManager
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(PluginManager));
+
+        internal static IReadOnlyCollection<IOTelExtension> TryLoadPlugins(JsonConfigurationSource pluginsConfig)
+        {
+            if (pluginsConfig == null)
+            {
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            var targetFramework = FrameworkDescription.Instance.TargetFramework;
+
+            Log.Debug("Executing plugins configuration: {0}", pluginsConfig);
+            Log.Information("Trying to load plugins with target framework '{0}'.", targetFramework);
+
+            string[] pluginFiles;
+
+            try
+            {
+                // TODO: Here additional metadata could be loaded (eg: for security and integrity)
+                // instead of just string path
+                pluginFiles = pluginsConfig.GetValue<JToken>($"['{targetFramework}']").ToObject<string[]>();
+            }
+            catch (ArgumentException ex)
+            {
+                Log.Warning(ex, "Could not parse list of plugin paths. Invalid plugin configuration provided.");
+
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            if (pluginFiles == null || !pluginFiles.Any())
+            {
+                Log.Information("Skipping plugins load. Could not find any plugins with target framework '{0}'.", targetFramework);
+
+                return ArrayHelper.Empty<IOTelExtension>();
+            }
+
+            var loadedPlugins = TryLoadPlugins(pluginFiles);
+
+            Log.Information("Successfully loaded '{0}' plugin(s).", property: loadedPlugins.Count);
+
+            return loadedPlugins;
+        }
+
+        private static IReadOnlyCollection<IOTelExtension> TryLoadPlugins(string[] pluginFiles)
+        {
+            var loaded = new List<IOTelExtension>();
+
+            foreach (string file in pluginFiles)
+            {
+                string fullPath = Path.GetFullPath(file);
+
+                if (File.Exists(fullPath))
```

Comment:
nit: the if is pretty big wrapping a try-catch a better if !file.Exists { log; continue; } makes the code easier to read.